Generate legacy annotations. Fixes #379

sa-bpelakh opened the pull request:
This will only `legacyAnnotations.ttl` via CONSTRUCT query to restore the old `rdfs:label`/`rdfs:comment` annotations. I don't know the right release notes verbiage or whether we should also have a JSON-LD and RDF/XML versions of this file. 

PS This requires onto_tool v0.10.0 or later (including the fix in https://github.com/semanticarts/ontology-toolkit/pull/47, which is not yet merged at this time).

uscholdm commented on bundle.yaml:
```diff
@@ -97,6 +97,32 @@ actions:
+- action: "transform"
+  tool: "serializer"
+  source: "{output}/legacyAnnotations.ttl"
+  target: "{output}/legacyAnnotations.ttl"
```
How can the source and target be the same file? 
As a sanity check, can I see a sample output of a run of this program?

sa-bpelakh replied:
The serializer doesn't seem to mind if you use the same file for input and output. I tested it, and I got the following output (just the first part included, the whole thing is over 1500 lines).
```
@prefix gist: <https://ontologies.semanticarts.com/gist/> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

gist:Account
        rdfs:label "Account"^^xsd:string ;
        rdfs:comment "An agreement having a balance, as in a bank account, or credit card account, or Accounts Receivable account."^^xsd:string ;
        .

gist:Actuator
        rdfs:label "Actuator"^^xsd:string ;
        rdfs:comment "A device that can affect the real world via a message interface"^^xsd:string ;
        .

gist:Address
        rdfs:label "Address"^^xsd:string ;
        rdfs:comment "A reference to a place (real or virtual) that can be located by some routing algorithm, and where messages or things can be sent to or retrieved from. E.
g. PO Box or URL to a PDF file."^^xsd:string ;
        .

gist:Agreement
        rdfs:label "Agreement"^^xsd:string ;
        rdfs:comment "Something which two or more People or Organizations mutually commit to do."^^xsd:string ;
        .

gist:Area
        rdfs:label "Area"^^xsd:string ;
        rdfs:comment "A measurement of two-dimensional space."^^xsd:string ;
        .

gist:AreaUnit
        rdfs:label "Area Unit"^^xsd:string ;
        rdfs:comment "A unit of two-dimensional area, such as square inches or hectares."^^xsd:string ;
        .
```

uscholdm replied:
OK - I'd like to have a quick peak at the whole output file so I can check for the more interesting cases.

uscholdm commented:
This loses a modest amount of information. For example:

```
gist:CoherentUnit
	rdfs:label "Coherent Unit"^^xsd:string ;
	rdfs:comment
		"A unit that is expressed in units that have no conversions.  It may be a simple unit.  It may also be a product or ratio unit that bottoms out in simple units."^^xsd:string ,
		"a simple unit: kilogram"^^xsd:string ,
		"coherent unit is the physics term for this, informally you might think of it as the standard unit for a given dimension."^^xsd:string ,
		"in principle, the CoherentUnit for a ProductUnit or RatioUnit can be inferred by recursively decomposing the products and ratios into their respective CoherentUnits, bottoming out in SimpleUnits"^^xsd:string ,
		"the standard unit for acceleration is meters per square second (feet per square second  requires a conversion)"^^xsd:string ,
		"CU"^^xsd:string 
		;
	.
```
This will be pretty confusing in Protege, say compared to:

```
gist:CoherentUnit
	rdfs:label "Coherent Unit"^^xsd:string ;
	rdfs:comment
		"DEFINITION: A unit that is expressed in units that have no conversions.  It may be a simple unit.  It may also be a product or ratio unit that bottoms out in simple units."^^xsd:string ,
		"EXAMPLE: a simple unit: kilogram"^^xsd:string,
		"EXAMPLE: the standard unit for acceleration is meters per square second (feet per square second  requires a conversion)"^^xsd:string ,
		"NOTE: coherent unit is the physics term for this, informally you might think of it as the standard unit for a given dimension."^^xsd:string ,
		"NOTE: in principle, the CoherentUnit for a ProductUnit or RatioUnit can be inferred by recursively decomposing the products and ratios into their respective CoherentUnits, bottoming out in SimpleUnits"^^xsd:string ,
                "ALTLABEL: CU"^^xsd:string ;
		;
	.
```

But frankly, it might not be worth the trouble. Users can always look at the original OWL.

uscholdm commented:
If you want to merge this, go ahead. We can always add enhancements later if we wish.

rjyounes reviewed (commented):
I agree with Michael's point that it would be best to have a faithful rendering of the original rdfs:comment. It should be an easy matter to do individual union clauses and some string transforms in the query based on which SKOS annotation it came from. If you agree that this can be done pretty quickly, I think it would be preferable to add it. Or at a minimum, uppercase the first letter of each string.

uscholdm commented:
Also: since these are not legacy annotatoins, they will always be up to date annotations, perhaps the file should be called rdfsAnnotations

sa-bpelakh commented:
@uscholdm I renamed the file as per your suggestion.
@rjyounes I have updated the query to retain the information, as follows:
```
gist:Category
        rdfs:label "Category"^^xsd:string ;
        rdfs:comment
                "DEFINITION: A concept or label used to categorize other instances informally. Things that can be thought of as types are usually Categories."^^xsd:string ,
                "EXAMPLE: Tags used in folksonomies; formal definitions from other systems."^^xsd:string ,
                "NOTE: Often a 'bucket' can be modeled either as an owl:Class or as a gist:Category. Use the latter if you don't care much about the formal structure of the different types, or if there is a whole hierarchy of types that are going to be managed by a group separate from the ontology developers. The formal structure may be defined elsewhere and linked to, if necessary."^^xsd:string
                ;
        .
```
@uscholdm was OK with it.

uscholdm commented:
Nicely done.

rjyounes commented:
Ah, I didn't think of doing it that way.
